pkg/gatewayserver/gatewayserver.go

```diff
@@ -505,7 +513,11 @@ func (gs *GatewayServer) Connect(ctx context.Context, frontend io.Frontend, ids
 		existingConnEntry.tasksDone.Wait()
 	}
 
-	registerGatewayConnect(ctx, ids, frontend.Protocol())
+	registerGatewayConnect(ctx, ids, frontend.Protocol(), &ttnpb.GatewayConnectionStats{
```

Comment:
Since you're passing the stats, you may drop the protocol parameter (since the function can use `stats.Protocol`).